configs/packageConfig.js

```diff
@@ -1,21 +1,99 @@
+const fs = require('fs');
+const path = require('path');
+const util = require('util');
+const crypto = require('crypto');
+
 const { Platform, build } = require('electron-builder');
 
+const readFileAsync = util.promisify(fs.readFile);
+const writeFileAsync = util.promisify(fs.writeFile);
+const mkdirAsync = util.promisify(fs.mkdir);
 const args = process.argv.slice(2);
 const targets = args && args.length ? args : null;
 
 if (!targets) {
   throw new Error("No arguments provided. Usage example: 'node ./packageConfig.js {mac/linux/windows}'");
 }
 
+const fileHashList = {
+  dmg: {
+    file: 'dmg_installer',
+    hash: 'dmg_sha512'
+  },
+  exe: {
+    file: 'exe_installer',
+    hash: 'exe_sha512'
+  },
+  deb: {
+    file: 'deb_installer',
+    hash: 'deb_sha512'
+  },
+  snap: {
+    file: 'snap_installer',
+    hash: 'snap_sha512'
+  },
+  AppImage: {
+    file: 'AppImage_installer',
+    hash: 'AppImage_sha512'
+  }
+};
+
 const nodeFiles = {
   mac: { from: 'node/mac/', to: 'node/' },
   windows: { from: 'node/windows/', to: 'node/' },
   linux: { from: 'node/linux/', to: 'node/' }
 };
 
+const artifactsToPublishFile = path.join(__dirname, '..', 'release', 'publishFilesList.json');
+
+const generateFileHashFile = async ({ destination }) => {
+  try {
+    var dirname = path.dirname(destination);
+    if (!fs.existsSync(dirname)) {
+      await mkdirAsync(dirname);
+    }
+    if (!fs.existsSync(destination)) {
+      await writeFileAsync(destination, JSON.stringify(fileHashList));
+    }
+  } catch (error) {
+    console.error(error);
+    process.exit(1);
+  }
+};
+
+generateFileHashFile({ destination: artifactsToPublishFile });
+
+const getFileHash = async ({ filename }) => {
+  const shasum = crypto.createHash('sha512');
+  const fileContent = await readFileAsync(filename);
+  const hashsum = shasum.update(fileContent);
+  const hash = hashsum.digest('hex');
+  return hash;
+};
+
+const compileHashListFile = async ({ artifactsToPublishFile, artifactPaths }) => {
+  const acceptedSuffixes = ['dmg', 'exe', 'deb', 'snap', 'AppImage'];
+  const fileContent = await readFileAsync(artifactsToPublishFile);
+  let hashList = JSON.stringify(JSON.parse(fileContent));
```

Comment:
[hint](https://lmgtfy.com/?q=nodejs+read+file+to+string)

Reply:
That's just disrespectful.